static/js/whyis.js

```diff
@@ -905,56 +905,43 @@ function whyis() {
             },
         };
     }]);
-    
-    app.factory("getLabel", ["$http", '$q', function($http, $q) {
-        var promises = {}
-        function getLabel(uri) {
-            if (getLabel.labels[uri] === undefined && promises[uri] === undefined) {
-                if (!promises[uri]) {
-                    var localPart = uri.split("#").filter(function(d) {return d.length > 0});
-                    localPart = localPart[localPart.length-1];
-                    localPart = localPart.split("/").filter(function(d) {return d.length > 0});
-                    localPart = localPart[localPart.length-1];
-                    getLabel.labels[uri] = localPart;
-                    promises[uri] = $q.defer();
-                    $http.get(ROOT_URL+'about?uri='+encodeURI(uri)+"&view=label")
-                        .then(function(data, status, headers, config) {
-                            if (status == 200) {
-                                var label = data.data;
-                                getLabel.labels[uri] = data.data;
-                                promises[uri].resolve(getLabel.labels[uri]);
-                            }
-                        });
-                }
-            }
-            return promises[uri].promise;
-        };
-        getLabel.labels = {};
-        return getLabel;
-    }]);
 
-    app.filter("label", ["$http", '$q','getLabel', function($http, $q, getLabel) {
-        function label(uri) {
-            if (getLabel.labels[uri] === undefined) {
-                var localPart = uri.split("#").filter(function(d) {return d.length > 0});
+    // Returns a promise that resolves to the label
+    app.factory("getLabel", ["$http", function($http) {
+	function getLabel(uri) {
+	    if(getLabel.labels[uri] === undefined) {
+		var localPart = uri.split("#").filter(function(d) {return d.length > 0});
                 localPart = localPart[localPart.length-1];
                 localPart = localPart.split("/").filter(function(d) {return d.length > 0});
                 localPart = localPart[localPart.length-1];
-                label.labels[uri] = {"label": localPart};
-                $http.get(ROOT_URL+'about?uri='+encodeURI(uri)+"&view=label")
-                    .then(function(data, status, headers, config) {
-                        if (status == 200) {
-                            var label = data.data;
-                            getLabel.labels[uri].label = data.data;
-                        }
-                    });
-            }
-            return getLabel.labels[uri];
```

Comment:
This is part of the filter code, but the filter was changed to return the label itself rather than an object.